<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1540</width>
    <height>838</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>USDB Syncer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLineEdit" name="lineEdit_search">
        <property name="placeholderText">
         <string>Search for anything ...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_update_hint">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;a href=&quot;https://github.com/bohning/usdb_syncer/releases&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#3586ff;&quot;&gt;USDB Syncer VERSION available!&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTableView" name="table_view">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>3</verstretch>
       </sizepolicy>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SelectionMode::ExtendedSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
      </property>
      <property name="sortingEnabled">
       <bool>true</bool>
      </property>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>20</number>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>20</number>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderMinimumSectionSize">
       <number>24</number>
      </attribute>
      <attribute name="verticalHeaderDefaultSectionSize">
       <number>24</number>
      </attribute>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="button_download">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Download</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/resources.qrc">
          <normaloff>:/icons/drive-download.png</normaloff>:/icons/drive-download.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="button_pause">
        <property name="text">
         <string>Pause</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/resources.qrc">
          <normaloff>:/icons/control-pause.png</normaloff>:/icons/control-pause.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="bar_download_progress">
        <property name="value">
         <number>0</number>
        </property>
        <property name="textVisible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_download_progress">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>0/0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_select_song_dir">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Select song folder</string>
        </property>
        <property name="icon">
         <iconset resource="../resources/resources.qrc">
          <normaloff>:/icons/folder_note.png</normaloff>:/icons/folder_note.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_song_dir">
        <property name="text">
         <string>./songs</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1540</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_tools">
    <property name="title">
     <string>Tools</string>
    </property>
    <property name="toolTipsVisible">
     <bool>true</bool>
    </property>
    <addaction name="action_settings"/>
    <addaction name="action_meta_tags"/>
    <addaction name="action_generate_song_pdf"/>
    <addaction name="action_generate_song_json"/>
   </widget>
   <widget class="QMenu" name="menu_usdb">
    <property name="title">
     <string>USDB</string>
    </property>
    <property name="toolTipsVisible">
     <bool>true</bool>
    </property>
    <addaction name="action_usdb_login"/>
    <addaction name="action_refetch_song_list"/>
   </widget>
   <widget class="QMenu" name="menu_about">
    <property name="title">
     <string>Help</string>
    </property>
    <property name="toolTipsVisible">
     <bool>true</bool>
    </property>
    <addaction name="action_show_log"/>
    <addaction name="action_about"/>
   </widget>
   <widget class="QMenu" name="menu_songs">
    <property name="title">
     <string>Songs</string>
    </property>
    <property name="toolTipsVisible">
     <bool>true</bool>
    </property>
    <widget class="QMenu" name="menu_custom_data">
     <property name="title">
      <string>Custom Data</string>
     </property>
     <property name="icon">
      <iconset resource="../resources/resources.qrc">
       <normaloff>:/icons/drawer.png</normaloff>:/icons/drawer.png</iconset>
     </property>
    </widget>
    <widget class="QMenu" name="menu_open_song_in">
     <property name="title">
      <string>Open Song in</string>
     </property>
     <property name="icon">
      <iconset resource="../resources/resources.qrc">
       <normaloff>:/icons/music--arrow.png</normaloff>:/icons/music--arrow.png</iconset>
     </property>
     <addaction name="action_open_song_in_usdx"/>
     <addaction name="action_open_song_in_vocaluxe"/>
     <addaction name="action_open_song_in_performous"/>
     <addaction name="separator"/>
     <addaction name="action_open_song_in_yass_reloaded"/>
     <addaction name="action_open_song_in_karedi"/>
     <addaction name="action_open_song_in_ultrastar_manager"/>
    </widget>
    <widget class="QMenu" name="menu_rate_song_on_usdb">
     <property name="title">
      <string>Rate Song on USDB</string>
     </property>
     <property name="icon">
      <iconset resource="../resources/resources.qrc">
       <normaloff>:/icons/rating.png</normaloff>:/icons/rating.png</iconset>
     </property>
     <addaction name="action_rate_1star"/>
     <addaction name="action_rate_2stars"/>
     <addaction name="action_rate_3stars"/>
     <addaction name="action_rate_4stars"/>
     <addaction name="action_rate_5stars"/>
    </widget>
    <addaction name="action_songs_download"/>
    <addaction name="action_songs_abort"/>
    <addaction name="action_show_in_usdb"/>
    <addaction name="action_post_comment_in_usdb"/>
    <addaction name="menu_rate_song_on_usdb"/>
    <addaction name="separator"/>
    <addaction name="action_open_song_folder"/>
    <addaction name="menu_open_song_in"/>
    <addaction name="menu_custom_data"/>
    <addaction name="action_pin"/>
    <addaction name="separator"/>
    <addaction name="action_delete"/>
   </widget>
   <widget class="QMenu" name="menu_local">
    <property name="title">
     <string>Local</string>
    </property>
    <property name="toolTipsVisible">
     <bool>true</bool>
    </property>
    <addaction name="action_find_local_songs"/>
    <addaction name="action_import_usdb_ids"/>
    <addaction name="action_export_usdb_ids"/>
   </widget>
   <widget class="QMenu" name="menu_view">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <addaction name="menu_view"/>
   <addaction name="menu_songs"/>
   <addaction name="menu_usdb"/>
   <addaction name="menu_local"/>
   <addaction name="menu_tools"/>
   <addaction name="menu_about"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dock_log">
   <property name="windowTitle">
    <string>Log</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QToolButton" name="toolButton_errors">
         <property name="toolTip">
          <string>Errors</string>
         </property>
         <property name="text">
          <string>Errors</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/icons/error.png</normaloff>:/icons/error.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButton_warnings">
         <property name="toolTip">
          <string>Warnings</string>
         </property>
         <property name="text">
          <string>Warnings</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/icons/warning.png</normaloff>:/icons/warning.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButton_infos">
         <property name="toolTip">
          <string>Information</string>
         </property>
         <property name="text">
          <string>Infos</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/icons/info.png</normaloff>:/icons/info.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButton_debugs">
         <property name="text">
          <string>Debugs</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/icons/bug.png</normaloff>:/icons/bug.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="plainTextEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dock_search">
   <property name="windowTitle">
    <string>Filters</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLineEdit" name="line_edit_search_filters">
       <property name="frame">
        <bool>false</bool>
       </property>
       <property name="placeholderText">
        <string>Search filters</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTreeView" name="search_view">
       <property name="frameShape">
        <enum>QFrame::Shape::NoFrame</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_meta_tags">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/tag-hash.png</normaloff>:/icons/tag-hash.png</iconset>
   </property>
   <property name="text">
    <string>Create Meta Tags</string>
   </property>
  </action>
  <action name="action_settings">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/cog.png</normaloff>:/icons/cog.png</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="action_generate_song_pdf">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/report.png</normaloff>:/icons/report.png</iconset>
   </property>
   <property name="text">
    <string>Generate Song List (PDF)</string>
   </property>
  </action>
  <action name="action_generate_song_json">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/report.png</normaloff>:/icons/report.png</iconset>
   </property>
   <property name="text">
    <string>Generate Song List (JSON)</string>
   </property>
  </action>
  <action name="action_refetch_song_list">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/check_for_update.png</normaloff>:/icons/check_for_update.png</iconset>
   </property>
   <property name="text">
    <string>Check USDB Song List</string>
   </property>
   <property name="toolTip">
    <string>Redownload the entire song list from USDB</string>
   </property>
  </action>
  <action name="action_find_local_songs">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/database.png</normaloff>:/icons/database.png</iconset>
   </property>
   <property name="text">
    <string>Find Local Songs</string>
   </property>
   <property name="toolTip">
    <string>Select songs matching a txt file in the given folder</string>
   </property>
  </action>
  <action name="action_show_log">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/log.png</normaloff>:/icons/log.png</iconset>
   </property>
   <property name="text">
    <string>Show Log</string>
   </property>
  </action>
  <action name="action_import_usdb_ids">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/document-export.png</normaloff>:/icons/document-export.png</iconset>
   </property>
   <property name="text">
    <string>Select Songs from Files</string>
   </property>
   <property name="toolTip">
    <string>Read song IDs from selected files and select the corresponding songs in the song table</string>
   </property>
  </action>
  <action name="action_export_usdb_ids">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/document-import.png</normaloff>:/icons/document-import.png</iconset>
   </property>
   <property name="text">
    <string>Export selected IDs to File</string>
   </property>
   <property name="toolTip">
    <string>Write the IDs of the selected songs to a file</string>
   </property>
  </action>
  <action name="action_songs_download">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/drive-download.png</normaloff>:/icons/drive-download.png</iconset>
   </property>
   <property name="text">
    <string>Download</string>
   </property>
  </action>
  <action name="action_usdb_login">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/faviconUSDB.png</normaloff>:/icons/faviconUSDB.png</iconset>
   </property>
   <property name="text">
    <string>USDB Login</string>
   </property>
  </action>
  <action name="action_about">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/app/appicon_128x128.png</normaloff>:/app/appicon_128x128.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="action_show_in_usdb">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/faviconUSDB.png</normaloff>:/icons/faviconUSDB.png</iconset>
   </property>
   <property name="text">
    <string>Show on USDB</string>
   </property>
  </action>
  <action name="action_post_comment_in_usdb">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/balloon.png</normaloff>:/icons/balloon.png</iconset>
   </property>
   <property name="text">
    <string>Post Comment on USDB</string>
   </property>
  </action>
  <action name="action_open_song_folder">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/folder_note.png</normaloff>:/icons/folder_note.png</iconset>
   </property>
   <property name="text">
    <string>Open Song Folder</string>
   </property>
  </action>
  <action name="action_delete">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/bin.png</normaloff>:/icons/bin.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="toolTip">
    <string>Move to system recycle bin</string>
   </property>
  </action>
  <action name="action_pin">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/pin.png</normaloff>:/icons/pin.png</iconset>
   </property>
   <property name="text">
    <string>Pin</string>
   </property>
   <property name="toolTip">
    <string>Disable or reenable updates to local files</string>
   </property>
  </action>
  <action name="action_songs_abort">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/minus-circle.png</normaloff>:/icons/minus-circle.png</iconset>
   </property>
   <property name="text">
    <string>Abort Download</string>
   </property>
  </action>
  <action name="action_delete_saved_search">
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_add_saved_search">
   <property name="text">
    <string>Save current search</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_update_saved_search">
   <property name="text">
    <string>Update with current search</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_set_saved_search_default">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Apply on startup</string>
   </property>
   <property name="toolTip">
    <string>If checked, this search will automatically applied every time you start the application.</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_set_saved_search_subscribed">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Download new matches</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_open_song_in_karedi">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/karedi.png</normaloff>:/icons/karedi.png</iconset>
   </property>
   <property name="text">
    <string>Karedi</string>
   </property>
  </action>
  <action name="action_open_song_in_performous">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/performous.png</normaloff>:/icons/performous.png</iconset>
   </property>
   <property name="text">
    <string>Performous</string>
   </property>
  </action>
  <action name="action_open_song_in_usdx">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/usdx.png</normaloff>:/icons/usdx.png</iconset>
   </property>
   <property name="text">
    <string>UltraStar Deluxe</string>
   </property>
  </action>
  <action name="action_open_song_in_ultrastar_manager">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/ultrastar-manager.png</normaloff>:/icons/ultrastar-manager.png</iconset>
   </property>
   <property name="text">
    <string>UltraStar Manager</string>
   </property>
  </action>
  <action name="action_open_song_in_vocaluxe">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/vocaluxe.png</normaloff>:/icons/vocaluxe.png</iconset>
   </property>
   <property name="text">
    <string>Vocaluxe</string>
   </property>
  </action>
  <action name="action_open_song_in_yass_reloaded">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/yass-reloaded.png</normaloff>:/icons/yass-reloaded.png</iconset>
   </property>
   <property name="text">
    <string>YASS Reloaded</string>
   </property>
  </action>
  <action name="action_rate_in_usdb">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icons/rating.png</normaloff>:/icons/rating.png</iconset>
   </property>
   <property name="text">
    <string>action_rate_in_usdb</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="action_rate_1star">
   <property name="text">
    <string>★</string>
   </property>
   <property name="toolTip">
    <string>unusable, needs a total rework</string>
   </property>
  </action>
  <action name="action_rate_2stars">
   <property name="text">
    <string>★★</string>
   </property>
   <property name="toolTip">
    <string>hardly usable, needs a lot of improvements</string>
   </property>
  </action>
  <action name="action_rate_3stars">
   <property name="text">
    <string>★★★</string>
   </property>
   <property name="toolTip">
    <string>usable, but room for improvement</string>
   </property>
  </action>
  <action name="action_rate_4stars">
   <property name="text">
    <string>★★★★</string>
   </property>
   <property name="toolTip">
    <string>room for some improvements</string>
   </property>
  </action>
  <action name="action_rate_5stars">
   <property name="text">
    <string>★★★★★</string>
   </property>
   <property name="toolTip">
    <string>simply perfect</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
